pkg/controller/build/reconciler.go

```diff
@@ -1188,12 +1187,35 @@ func (b *buildReconciler) syncMachineOSBuild(ctx context.Context, mosb *mcfgv1.M
 			return fmt.Errorf("could not get MachineConfigPool from MachineOSBuild %q: %w", mosb.Name, err)
 		}
 
-		// An mosb which had previously been forgotten by the queue and is no longer desired by the mcp should not build
 		if mosb.ObjectMeta.Labels[constants.RenderedMachineConfigLabelKey] != mcp.Spec.Configuration.Name {
 			klog.Infof("The MachineOSBuild %q which builds the rendered Machine Config %q is no longer desired by the MCP %q", mosb.Name, mosb.ObjectMeta.Labels[constants.RenderedMachineConfigLabelKey], mosb.ObjectMeta.Labels[constants.TargetMachineConfigPoolLabelKey])
 			return nil
 		}
 
+		oldMCName := mcp.Status.Configuration.Name
+		newMCName := mcp.Spec.Configuration.Name
```

Comment:
Instead of having separate variables here for the MC names, can we just use `mcp.Status.Configuration.Name` and `mcp.Spec.Configuration.Name` in the `machineConfigLister.Get()` calls below?
I think that would make it a bit more cleaner, but just a suggestion :)